android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -0,0 +1,400 @@
+      }
+      mAnimationsManager.progressLayoutAnimation(view.getId(), preparedValues, true);
+      viewSourcePreviousSnapshot.originY = originY;
+      mCurrentSharedTransitionViews.remove(view.getId());
+    }
+    if (mCurrentSharedTransitionViews.isEmpty()) {
+      mSharedTransitionParent.clear();
+      mSharedTransitionInParentIndex.clear();
+      if (mTransitionContainer != null) {
+        ViewParent transitionContainerParent = mTransitionContainer.getParent();
+        if (transitionContainerParent != null) {
+          ((ViewGroup) transitionContainerParent).removeView(mTransitionContainer);
+        }
+      }
+      mIsSharedTransitionActive = false;
+    }
+  }
+
+  @Nullable
+  private View findScreen(View view) {
+    ViewParent parent = view.getParent();
+    while (parent != null) {
+      if (parent.getClass().getSimpleName().equals("Screen")) {
+        return (View) parent;
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  @Nullable
+  private View findStack(View view) {
+    ViewParent parent = view.getParent();
+    while (parent != null) {
+      if (parent.getClass().getSimpleName().equals("ScreenStack")) {
+        return (View) parent;
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  protected void makeSnapshot(View view) {
+    Snapshot snapshot = new Snapshot(view);
+    View screen = findScreen(view);
+    if (screen != null) {
+      snapshot.originY -= screen.getTop();
+    }
+    mSnapshotRegistry.put(view.getId(), snapshot);
+  }
+
+  protected void visitTreeForTags(int[] viewTags, boolean clearConfig) {
+    if (viewTags == null) {
+      return;
+    }
+    for (int viewTag : viewTags) {
+      visitTree(viewTag, clearConfig);
+    }
+  }
+
+  private void visitTree(int tag, boolean clearConfig) {
+    if (tag == -1) {
+      return;
+    }
+    ViewGroup viewGroup;
+    ViewGroupManager<ViewGroup> viewGroupManager;
+    ReanimatedNativeHierarchyManager reanimatedNativeHierarchyManager =
+        mAnimationsManager.getReanimatedNativeHierarchyManager();
+    try {
+      View view = reanimatedNativeHierarchyManager.resolveView(tag);
+      if (!clearConfig && mAnimationsManager.hasAnimationForTag(tag, "sharedElementTransition")) {
+        mRemovedSharedViews.add(view);
+        makeSnapshot(view);
+      }
+      if (clearConfig) {
+        mNativeMethodsHolder.clearAnimationConfig(tag);
+      }
+
+      if (!(view instanceof ViewGroup)) {
+        return;
+      }
+      viewGroup = (ViewGroup) view;
+      viewGroupManager =
+          (ViewGroupManager) reanimatedNativeHierarchyManager.resolveViewManager(tag);
+    } catch (IllegalViewOperationException e) {
+      return;
+    }
+    if (viewGroupManager == null) {
+      return;
+    }
+    for (int i = 0; i < viewGroupManager.getChildCount(viewGroup); i++) {
+      View child = viewGroupManager.getChildAt(viewGroup, i);
+      visitTree(child.getId(), clearConfig);
```

Comment:
We call `getId()` on child, only to then use the tag to get back the view. Wouldn't it be better to instead take the `View` as argument?